elastic/kibana
# [Upgrade Assistant] Overview page redesign

Fixes: https://github.com/elastic/kibana/issues/79208

## Summary
Improve Upgrade Assistant overview page with more comprehensive instructions, documentation links and add links to observability and discovery to allow users to better understand their deprecation logs.

![Screenshot 2021-08-11 at 15 32 58](https://user-images.githubusercontent.com/1191206/129038453-dbbb7678-5060-4c36-b08a-ec5c19579d57.png)

### Upper limit for review logs step
![Screenshot 2021-08-17 at 13 39 12](https://user-images.githubusercontent.com/1191206/129719474-2dc5049b-5046-4cd3-8cdd-e367eba4ed07.png)

### Error for review logs step stats panel
![Screenshot 2021-08-17 at 13 50 34](https://user-images.githubusercontent.com/1191206/129722963-acf79a73-a6f5-4bd5-88b1-141b7321bd94.png)

### Just one stat shown for review logs step stats panel
![Screenshot 2021-08-17 at 14 06 50](https://user-images.githubusercontent.com/1191206/129722804-58f3fa6b-f822-4bef-b752-8401e9df43fb.png)


#### How to test
1. To enable UA, set `xpack.upgrade_assistant.readonly: false` in your `kibana.yml`
2. Start elasticsearch with `yarn es snapshot --license=trial` 
3. Start kibana with `yarn start`
4. Navigate to UA and enable the toggle to start collecting logs
6. In order to see logs, navigate around kibana for a bit and then come back to UA and you should be able to see a few logs in the Discovery or Observability apps.

##### How to test cloud related changes
When running Kibana locally add following values to `kibana.yml` file to simulate Cloud environment

```
xpack.cloud.id: 'remote-deployment:dXMtY2VudHJhbDEuZ2NwLm'
xpack.cloud.base_url: 'https://staging.found.no/deployments/d74e0a97f269463ca2ec05dcbe703662'
```

##### How to test toggle error messages

* **Initial load error**:  If you edit [this line](https://github.com/elastic/kibana/blob/545c1420c285af8f5eee56f414bd6eca735aea11/x-pack/plugins/upgrade_assistant/public/application/lib/api.ts#L70) locally and replace `deprecation_logging` with `fake_deprecation_logging` it should return a 404 error.
* **Update load error**:  If you edit [this line](https://github.com/elastic/kibana/blob/545c1420c285af8f5eee56f414bd6eca735aea11/x-pack/plugins/upgrade_assistant/public/application/lib/api.ts#L77) locally and replace `deprecation_logging` with `fake_deprecation_logging` it should return a 404 error.
